docs/aws/audit/elbmonitoring/rules/classic_elb_security_group_remediation.mdx

```diff
@@ -1,6 +1,206 @@
+   elb_name = 'your-elb-name'
+   default_sg_id = 'sg-0123456789abcdef0'  # Replace with your default security group ID
+   periodic_check(elb_name, default_sg_id)
+   ```
+
+By following these steps, you can ensure that your ELB always has at least one valid security group assigned, thus preventing the misconfiguration.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Log in to the AWS Management Console and open the Amazon EC2 dashboard at https://console.aws.amazon.com/ec2/.
+
+2. In the navigation pane, under LOAD BALANCING, click on Load Balancers.
+
+3. Select the Elastic Load Balancer (ELB) that you want to inspect.
+
+4. In the bottom panel, click on the "Security" tab. Here, you can see the security groups associated with the selected ELB.
+
+5. Check if there is at least one valid security group associated with the ELB. If there are no security groups or if the associated security groups are invalid, then it is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the Elastic Load Balancer (ELB) resources.
+
+2. Once the AWS CLI is set up, you can list all the load balancers in your account using the following command:
+
+   ```
+   aws elbv2 describe-load-balancers
+   ```
+
+   This command will return a JSON output with details about all the load balancers in your account.
+
+3. To check the security groups associated with each load balancer, you can use the following command:
+
+   ```
+   aws elbv2 describe-load-balancer-attributes --load-balancer-arn <load-balancer-arn>
+   ```
+
+   Replace `<load-balancer-arn>` with the ARN of the load balancer you want to check. This command will return a JSON output with the attributes of the specified load balancer, including the associated security groups.
+
+4. To check if the security group is valid, you can use the following command:
+
+   ```
+   aws ec2 describe-security-groups --group-ids <security-group-id>
+   ```
+
+   Replace `<security-group-id>` with the ID of the security group you want to check. This command will return a JSON output with the details of the specified security group. If the security group does not exist or is not associated with the load balancer, the command will return an error.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary AWS SDK for Python (Boto3) if you haven't done so already. You can do this by running the command `pip install boto3` in your terminal.
+
+2. Import the necessary modules and establish a session with AWS using your credentials. Here's a basic setup:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>